src/provideInlineCompletionItems.ts

```diff
@@ -0,0 +1,71 @@
+import * as vscode from "vscode";
+import { AutocompleteResult, ResultEntry } from "./binary/requests/requests";
+import { completionIsAllowed } from "./provideCompletionItems";
+import runCompletion from "./runCompletion";
+import { COMPLETION_IMPORTS } from "./selectionHandler";
+
+const INLINE_REQUEST_TIMEOUT = 3000;
+
+export default async function provideInlineCompletionItems(
+  document: vscode.TextDocument,
+  position: vscode.Position
+): Promise<vscode.InlineCompletionList> {
+  try {
+    if (!completionIsAllowed(document, position)) {
+      return new vscode.InlineCompletionList([]);
+    }
+
+    const isEmptyLine = document.lineAt(position.line).text.trim().length === 0;
+
+    const response = await runCompletion(
+      document,
+      position,
+      isEmptyLine ? INLINE_REQUEST_TIMEOUT : undefined
+    );
+
+    const completions = response?.results.map(
+      (result) =>
+        new vscode.InlineCompletionItem(
+          result.new_prefix,
+          calculateRange(position, response, result),
+          getAutoImportCommand(result, response, position)
+        )
+    );
+
+    return new vscode.InlineCompletionList(completions || []);
+  } catch (e) {
+    console.error(`Error setting up request: ${e}`);
+
+    return new vscode.InlineCompletionList([]);
+  }
+}
+
+function getAutoImportCommand(
+  result: ResultEntry,
+  response: AutocompleteResult,
+  position: vscode.Position
+): vscode.Command {
+  return {
+    arguments: [
+      {
+        currentCompletion: result.new_prefix,
+        completions: response.results,
+        position,
+        limited: response?.is_locked,
+      },
+    ],
+    command: COMPLETION_IMPORTS,
+    title: "accept completion",
+  };
+}
+
+function calculateRange(
```

Comment:
Now that binary version 4.0.36 is released its worth checking that snippets are working - this version includes the correct suffixes calculation.